homeassistant/components/http/web_runner.py

```diff
@@ -0,0 +1,64 @@
+"""HomeAssistant specific aiohttp Site."""
+import asyncio
+from ssl import SSLContext
+from typing import List, Optional, Union
+
+from aiohttp import web
+from yarl import URL
+
+
+class HomeAssistantTCPSite(web.BaseSite):
+    """HomeAssistant specific aiohttp Site.
+
+    Vanilla TCPSite accepts only str as host. However, the underlying asyncio's
+    create_server() implementation does take a list of strings to bind to multiple
+    host IP's. To support multiple server_host entries (e.g. to enable dual-stack
+    explicitly), we would like to pass an array of strings. Bring our own
+    implementation inspired by TCPSite.
+    """
+
+    __slots__ = ("_host", "_port", "_reuse_address", "_reuse_port", "_hosturl")
+
+    def __init__(
+        self,
+        runner: "web.BaseRunner",
+        host: Union[Optional[str], List[str]],
```

Comment:
This looks a little nicer imo

```suggestion
        host: Union[None, str, List[str]],
```